src/app/services/data.service.ts

```diff
@@ -123,6 +126,176 @@ export class DataService {
     }
     return Object.keys(result).sort().reverse();
   }
+  
+  public getRoutesLast30Days(): Route[] {    
+    var routesLast30Days = [];
+    const routes = this.getRoutes();
+    //TO DO: use object instead of array
+    const last30DaysStrings = this.getLast30DaysStrings();
+    for (var date of last30DaysStrings) {   
+      for (var route of routes) {
+        var routeDate = route.time.split(' ')[0];
+        if (date === routeDate) {
+          routesLast30Days.push(route);
+        }
+      }
+    }
+    return routesLast30Days;
+  }
+  public getRoutesLast6MonthsInclCurrent(): Route[] {    
+    var routesLast6Months = [];
+    const routes = this.getRoutes();
+    //TO DO: use object instead of array
+    const last6MonthsStrings = this.getLastXMonthsStrings(6);
+    for (var month of last6MonthsStrings) {   
+      for (var route of routes) {
+        var routeMonth = route.time.substring(0,7);
+        if (month === routeMonth) {
+          routesLast6Months.push(route);
+        }
+      }
+    }
+    return routesLast6Months;
+  }
+
+  public getDateString(date: Date): string {    
+    const dateStr = date.getFullYear()+'-'+formatNumber((date.getMonth()+1), 'en_US', '2.0-0')+'-'+formatNumber(date.getDate(), 'en_US', '2.0-0');
+    return dateStr;
+  }
+  
+  public getMonthString(date: Date): string {    
+    const monthStr = date.getFullYear()+'-'+formatNumber((date.getMonth()+1), 'en_US', '2.0-0');
+    return monthStr;
+  }
+
+  public getLast30DaysStrings(): string[] {
+    const today = Date.now();
+    const days = 24*3600*1000;
+    var last30DaysStrings = [];
```

Comment:
just started seeing this now: never use var
if it is a constant use 'const' otherwise use 'let'

applies to all files

Reply:
ok, but in this case the string array needs to be let, right?
cause it needs to be altered i.e. filled... right?

Reply:
if it is only filled no, but if it is replaced (something like last30DaysStrings = [] again) then let

Reply:
a little confusing... :D

anyway i replaced all var with let or const now in all files i edited. hope it's fine like that now. push coming...